### DISCUSSION TITLE
Possible to assign different OpenAI API connection per user to track external litellm usage??

### DISCUSSION THREAD

**Discussion by corticalstack** [open-webui/open-webui#1851] [Q&A]
Hi,

My LLM stack goes like this: Frontend open-webUI OpenAI API conenction > external litellm > models served by local ollama/cloud OpenAI/Groq.

On the external litellm I'd like to be able to track model usage per user of open-webui. The only way I can see how this can be tracked is to have a different OpenAI API connection with a different API key per user.  You could imagine, for Enterprise, mutliple users using open-webui, with an OpenAI API connection set up per team, to collect usage stats, enable chargeback of costs etc.

However, I don't believe it's possible to choose from a list of open-webui configured OpenAI API connections to assign to a user? Would love to be corrected or any other comments.

Thanks

**Comment by justinh-rahb** [comment]
There was a user in our Discord that contributed [this tutorial to the docs site](https://docs.openwebui.com/tutorial/langfuse). I remember him mentioning something about how to track by individual user. If you're on the Discord as well, mention this and I'll try to connect you two.

**Comment by Michelklingler** [comment]
I did not manage to track the users.
We will need to pass the user_ID variable to LiteLLM along with the Message_ID, message and model name.

**Reply by corticalstack** [reply]
Thanks @justinh-rahb for the links and reach out.

@Michelklingler Thanks for looking into it previously. Another alternative could be multiple OpenAI API connections, each with diff API key, to distinguish users/teams?

**Comment by skrenes** [comment]
"LiteLLM support has been deprecated". It seems like all the components for per-user API support is there. The "Connections" setting has support to add OpenAI API keys. It would be nice if we could add as many keys as we'd like and give them a friendly name. If the "Dashboard" then had an dropdown to select the key, this would be amazing. Not sure if there's a place to make feature requests.
